de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/v1_9/fsk2metadata/FSK2MetadataNodeModel.java

```diff
@@ -78,10 +78,13 @@ protected FSK2MetadataNodeModel() {
     final DataCell mathCell = createJSONCell(modelMath);
     final DataCell scriptCell = StringCellFactory.create(inObj.getModel());
     final DataCell visCell = StringCellFactory.create(inObj.getViz());
+    final DataCell pathCell = StringCellFactory
+        .create(inObj.getEnvironmentManager().get().getEnvironment().get().toString());

```

Comment:
Please check the environment manager and environment. They are optional and if missing this will produce NPE. Also the environment folder should at least be deleted on shutdown. A hook can be added to remove it on shutdown (*deleteOnExit*).
Ideally, the environment could be passed as JSON and restored in the node that actually uses it. That way the life of the folder could be limited to that node execution and it would not be necessary to remove it on shutdown. Anyway fixing the NPEs and removing the folder on shutdown should be enough for now.

Reply:
This is exactly what is happening, the path is passed to the JS, and used by the node afterward.